### PR TITLE
Improve documentation on state::Prune and Merkle state

### PR DESCRIPTION
**Pull request by peterschwarz**

### CODE REVIEW

**Review by agunde406 — APPROVED** [review]

File: libtransact/src/state/merkle/mod.rs [hunk 1]
```diff
@@ -17,6 +17,48 @@
...
+//! successors a given root is.
+//!
+//! * If state root has successors, then all of the nodes that were removed during the creation of
+//!   the root are removed. In other words, any data that would not be accessible from the given
+//!   state root from prior state roots would be removed.
+//! * If a state root has no successors (it's the tip of a successor chain), then all of the nodes
+//!   that were added by that state root are removed. In other words, any data that would not be
+//!   accessible by any other state root that still exists in the tree.
+//!
+//! The later method exists in services of removing forks in a successor chain of a tree.
```
Comment (jsmitchell) [inline libtransact/src/state/merkle/mod.rs:60]: s/services/service/

File: libtransact/src/state/merkle/mod.rs [hunk 2]
```diff
@@ -17,6 +17,48 @@
...
+//!
+//! Transact provides an addressable Merkle-Radix tree to store data for state. Let’s break that
+//! down: The tree is a Merkle tree because it is a copy-on-write data structure which stores
+//! successive node hashes from leaf-to-root upon any changes to the tree. For a given set of state
+//! transitions, we can generate a single root hash which points to that version of the tree.
+//!
+//! The root hash can be used to verify that the same set of state transitions applied against the
+//! same state ID in another database will produce the same results.
+//!
+//! For more information about general concepts, see the [Merkle]() page on Wikipedia.
```
Comment (jsmitchell) [inline libtransact/src/state/merkle/mod.rs:32]: Maybe strike these wikipedia references given this is code doc

File: libtransact/src/state/merkle/mod.rs [hunk 3]
```diff
@@ -17,6 +17,48 @@
...
+//!
+//! ## Radix Addresses
+//!
+//! The tree is an addressable Radix tree because addresses uniquely identify the paths to leaf
+//! nodes in the tree where information is stored. An address is a hex-encoded byte array.  In the
+//! tree implementation, each byte is a Radix path segment which identifies the next node in the
+//! path to the leaf containing the data associated with the address. This gives the tree a branch
+//! factor of 256.
+//!
+//! For more information about general concepts, see the Radix page on Wikipedia.
```
Comment (jsmitchell) [inline libtransact/src/state/merkle/mod.rs:42]: Strike wikipedia reference

**Review by jsmitchell — CHANGES_REQUESTED** [review]

File: libtransact/src/state/merkle/mod.rs (hunk 1 above)
Comment (peterschwarz) [reply libtransact/src/state/merkle/mod.rs]: Fixed

File: libtransact/src/state/merkle/mod.rs (hunk 2 above)
Comment (peterschwarz) [reply libtransact/src/state/merkle/mod.rs]: Removed

File: libtransact/src/state/merkle/mod.rs (hunk 3 above)
Comment (peterschwarz) [reply libtransact/src/state/merkle/mod.rs]: Removed

**Review by jsmitchell — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**